x/evm/keeper/msg_server.go

```diff
@@ -369,7 +369,7 @@ func (s msgServer) ConfirmChain(c context.Context, req *types.ConfirmChainReques
 		return nil, fmt.Errorf("'%s' has not been added yet", req.Name)
 	}
 
-	seqNo := s.snapshotter.GetLatestCounter(ctx)
+	seqNo := s.snapshotter.GetLatestCounterByRole(ctx, tss.MasterKey)
```

Comment:
I'd rather a new snapshot be taken every single time instead of using whatever lastest. `GetLatestCounterByRole` makes our state harder to manage and less elegant.

Reply:
Done :)